app/client/keybase/slip_test.go

```diff
@@ -2,225 +2,213 @@ package keybase
 
 import (
 	"encoding/hex"
+	"fmt"
+	"testing"
+
 	"github.com/pokt-network/pocket/shared/crypto"
 	"github.com/stretchr/testify/require"
-	"testing"
 )
 
 const (
+	// Test Vectors
+	testVector1SeedHex = "000102030405060708090a0b0c0d0e0f"
+	testVector2SeedHex = "fffcf9f6f3f0edeae7e4e1dedbd8d5d2cfccc9c6c3c0bdbab7b4b1aeaba8a5a29f9c999693908d8a8784817e7b7875726f6c696663605d5a5754514e4b484542"
+
 	// SLIPS-0010
+	testSeedHex       = "045e8380086abc6f6e941d6fe47ca93b86723bc246ec8c4beee411b410028675"
 	testChildAddrIdx1 = "8b83d7057df7ac1d20a2f0aa0edadf206eb6764d"
 )
 
-// https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-1-for-ed25519
-func TestSlip_DeriveChild_Vector1(t *testing.T) {
-	seed, err := hex.DecodeString("000102030405060708090a0b0c0d0e0f")
-	require.NoError(t, err)
-
-	type args struct {
-		path string
-		seed []byte
-	}
+func TestSlip_DeriveChild_TestVectors(t *testing.T) {
 	tests := []struct {
 		name        string
-		args        args
+		path        string
+		seed        string
 		wantPrivHex string
 		wantPubHex  string
 
 		wantErr bool
 	}{
+		// https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-1-for-ed25519
 		{
-			name: "Key(m) – master key",
-			args: args{
-				path: "m",
-				seed: seed,
-			},
+			name:        "TestVector1 Key derivation is deterministic for path `m` (master key)",
+			path:        "m",
+			seed:        testVector1SeedHex,
 			wantPrivHex: "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7",
 			wantPubHex:  "00a4b2856bfec510abab89753fac1ac0e1112364e7d250545963f135f2a33188ed",
-
-			wantErr: false,
+			wantErr:     false,
 		},
 		{
-			name: "Key(m/0')",
-			args: args{
-				path: "m/0'",
-				seed: seed,
-			},
+			name:        "TestVector1 Key derivation is deterministic for path `m/0'`",
```

Comment:
For my personal understanding, since th epath is `m/purpose'/coint_type'/child_idx'`, what is this deriving?

Reply:
@Olshansk So what we actually do to derive the key is first derive a master key `m/` then for each segment derive a child - for our use case this means we derive 3 keys before we actually get to the indexed derivation `m(key 1)/44'(key 2)/635'(key 3)/%d'(child key index)` this is the idea behind the of BIP-44. So we can change the purpose (`44'`) part of the path and with the same seed generate new accounts. For our use case this probably wont be the case but that is what we are actually doing. Many layers to the child 🧅

Reply:
Got it. So hypothetically (we don't need to do it), an application that wants to maintain privacy by generating lots of different receiving addresses could cache m/44/635 and only update the index, right?

Reply:
@Olshansk so If someone really wanted they could store the seed of the key generated from `m/44'/635'` and this is equivalent to the `SecretKey` of the `HMAC` key. Then they could derive the indexed child directly from this - however our current setup means this wouldn't work as we force the use of a `BIP-44` path when deriving children - this means if they saved `m/44'/635'` they couldn't use the path `193872'` to get that child as this is not valid if they used `m/44'/635'/193872'` as the path at this point they would not get the right child as they would re-generate the other "layers" of the child key. I hope this makes sense.

It would be possible to change this but I don't think the current use case requires this - maybe in a follow up PR or something